### PR TITLE
Prepare for 6.x

### PR DESCRIPTION
**Pull request by denisbr**
Include a lot of outstanding packaging improvements and fixes.

1. Unified varnishreload script (#70)
2. Unified systemd varnish.service unit file
3. Use network-online.target (#87)
4. Remove the default ``-T`` and ``-S`` parameters (#84)
5. Make ``TasksMax=infinity`` default for all distros

### CODE REVIEW

**Comment by Dridi** [comment]
We should also no longer create the `/ect/varnish/secret` file at packaging time. Still browsing the diff.

File: common/varnish.service [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+LimitNOFILE=131072
+
+# Locked shared memory - should suffice to lock the shared memory log
+# (varnishd -l argument)
+# Default log size is 80MB vsl + 1M vsm + header -> 82MB
+# unit is bytes
+LimitMEMLOCK=85983232
+
+# On systemd >= 228 enable this to avoid "fork failed" on reload.
+#TasksMax=infinity
```
Comment (Dridi) [inline common/varnish.service]: Is it a problem to enable it on systemd < 228?

File: debian/varnish.init [hunk 2]
```diff
@@ -88,7 +88,7 @@ reload_varnishd() {
     log_daemon_msg "Reloading $DESC" "$SERVICE"
     # Workaround for the reload script to make it work with instances
     export DAEMON_OPTS
-    if /usr/share/varnish/reload-vcl -q; then
+    if /usr/share/varnish/varnishreload -q; then
```
Comment (Dridi) [inline debian/varnish.init]: `varnishreload` doesn't have a `-q` option.

**Comment by Dridi** [comment]
I'm done browsing the diff, I hope I haven't missed anything else: LAGTM.

File: common/varnish.service (hunk 1 above)
Comment (denisbr) [reply common/varnish.service]: I'm not sure, and I have little prior knowledge about the original issue, I'm researching this now. For the record, we have these main targets:

```
debian-8-amd64
systemd 215

debian-9-amd64
systemd 232

centos-7-amd64
systemd 219

ubuntu-16.04-amd64
systemd 229
```

Comment (denisbr) [reply common/varnish.service]: Tested on debian jessie (systemd 215) and centos7 (systemd 219), which I believe to be the only distros in our targets that have systemd < 228. No immediate issues there. 
I think we should try adding this as a default.

File: debian/varnish.init (hunk 2 above)
Comment (denisbr) [reply debian/varnish.init]: I fixed this, thanks.

**Comment by denisbr** [comment]
@Dridi thanks for spotting the VARNISH_ADMIN_LISTEN_* remnants, purged those too now.

**Comment by Dridi** [comment]
I'm going to browse the source tree now that all is taken care of, in case I find something noteworthy.

File: redhat/varnish.sysconfig [hunk 3]
```diff
@@ -19,8 +19,7 @@ NPROCS="unlimited"
 
 # Init script support to reload/switch vcl without restart.
 # To make this work, you need to set the following variables
-# explicit: VARNISH_VCL_CONF, VARNISH_ADMIN_LISTEN_ADDRESS,
-# VARNISH_ADMIN_LISTEN_PORT, VARNISH_SECRET_FILE.
+# explicit: VARNISH_VCL_CONF, VARNISH_SECRET_FILE.
```
Comment (Dridi) [inline redhat/varnish.sysconfig]: VARNISH_SECRET_FILE spotted :p

File: redhat/varnish.params [hunk 4]
```diff
@@ -15,10 +15,6 @@ VARNISH_VCL_CONF=/etc/varnish/default.vcl
 # VARNISH_LISTEN_ADDRESS=192.168.1.5
 VARNISH_LISTEN_PORT=6081
 
-# Admin interface listen address and port
-VARNISH_ADMIN_LISTEN_ADDRESS=127.0.0.1
-VARNISH_ADMIN_LISTEN_PORT=6082
-
 # Shared secret file for admin interface
 VARNISH_SECRET_FILE=/etc/varnish/secret
```
Comment (Dridi) [inline redhat/varnish.params]: VARNISH_SECRET_FILE spotted :p

**Comment by Dridi** [comment]
Old reload script still in use:

https://github.com/denisbr/pkg-varnish-cache/blob/5d4713d/redhat/varnish.initrc#L29

**Comment by Dridi** [comment]
On the Debian side:

- The changelog looks unmaintained, maybe it should have a single entry like in the RPM spec?
- https://github.com/denisbr/pkg-varnish-cache/blob/6.x/debian/varnish.postinst#L63 still relevant ?

**Comment by Dridi** [comment]
This should say `python >= 2.7` to satisfy `vmodtool.py` and `vsctool.py`:

https://github.com/denisbr/pkg-varnish-cache/blob/e4d212120a649460a3d58642bdf0ef2cb1bbae79/redhat/varnish.spec#L80

**Comment by gquintard** [comment]
@rezan, which one?

**Comment by denisbr** [comment]
@Dridi Re python >= 2.7, did you consider the whole https://github.com/denisbr/pkg-varnish-cache/blob/e4d212120a649460a3d58642bdf0ef2cb1bbae79/redhat/varnish.spec#L96-L102 mess? I thought this still applied?

**Comment by Dridi** [comment]
@denis: @rezan was referring to the secret file. No MSE for varnish cache to my knowledge, so I'm against reverting that change.

Regarding Python >= 2.7 we need EPEL for jemalloc, and Python 3 is available via both EPEL 6 and 7:

https://apps.fedoraproject.org/packages/python34

**Comment by Dridi** [comment]
@denisbr one more python thing please, we also need it as a `BuildRequires`, and we may need to tweak the other dependencies in order to be package-name-agnostic as well. I guess at this point someone should try el6 builds and see if anything blows up.

**Comment by rezan** [comment]
Sorry, disregard my comment, I mixed up the repos/project :)

**Comment by denisbr** [comment]
Tried some builds, and eventually we got these packages built:

1. EL6
1. EL7
1. Debian Wheezy
1. Debian Jessie
1. Debian Stretch
1. Ubuntu Trusty
1. Ubuntu Xenial

Of these we're likely to ditch Debian Wheezy for 6.0 anyway.

**Comment by Dridi** [comment]
@denisbr once you're back to this, please add `KillMode=process` to `varnish.service` as you suggested on IRC, we definitely don't want the default behavior.

**Comment by Dridi** [comment]
I'm done with all the changes we were supposed to take care of, except one that we will discuss on Monday. I messed up commit messages on a couple key patches in my haste, so before any merge happens I'd like to see more testing and get an occasion to amend messages.

I would really appreciate a review of the last batch of commits: bfa631f...290c654.

**Pull request merged**